### DISCUSSION TITLE
How to send message through socketio when executing long time request?

### DISCUSSION THREAD

**Discussion by jesses2025smith** [fastapi/fastapi#10613] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request
import socketio
import time

sio = socketio.AsyncServer(async_mode="asgi", cors_allowed_origins="*")
app = FastAPI()

app.mount("/sio", socketio.ASGIApp(sio, socketio_path="socket.io"))

@app.get("/")
async def index():
    time.sleep(1)
    await sio.emit("message", "message")
    time.sleep(1)
    return {"hello": "world"}
```

### Description

it should recieve socketio message before "return" code if we connect the socketio server.

### Operating System

Linux

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.11.5

**Comment by YuriiMotov** [comment]
Don't use blocking `time.sleep(1)` inside async function. Use `await asyncio.sleep(1)` instead
